src/impl/graph.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef IMPL_GRAPH_HPP
+#define IMPL_GRAPH_HPP
+
+#include <edge.hpp>
+
+#include <map>
+#include <set>
+#include <utility>
+
+namespace graph_impl {
+
+template<typename Node, typename GraphTraitList, typename EdgeTraitList>
+class Graph;
+
+template<
```

Comment:
OK, but in my view, this makes a code difficult to understand. I would prefer to pass template parameters for Edge separately.

Reply:
The idea is it's hard to separate edge and graph traits from mathematics point of view.

Reply:
It is your choice, I just made my opinion :)